### PR TITLE
Fix the error page given when try the t/carbon.super as tenant

### PR DESCRIPTION
**Pull request by GANGANI**
### Proposed changes in this pull request
Resolves: https://github.com/wso2-enterprise/asgardeo-product/issues/2175

### Follow up actions
The **page_not_found.html** file needs to be added to _<IS_HOME>/repository/resources/identity/pages_ directory.

### CODE REVIEW

File: components/org.wso2.carbon.identity.context.rewrite.valve/src/main/java/org/wso2/carbon/identity/context/rewrite/valve/TenantContextRewriteValve.java [hunk 1]
```diff
@@ -273,17 +280,47 @@ private void handleInvalidTenantDomainErrorResponse(Response response, int error
...
+        try {
+            Path pageNotFoundHtmlResponse =
+                    Paths.get(CarbonUtils.getCarbonHome(), "repository", "resources", "identity", "pages",
+                            "page_not_found.html");
+            if (!Files.exists(pageNotFoundHtmlResponse) ||
+                    !Files.isRegularFile(pageNotFoundHtmlResponse)) {
+                if (log.isDebugEnabled()) {
+                    log.debug("pageNotFoundHtmlResponse is not present at: " + pageNotFoundHtmlResponse);
+                }
+            }
```
Comment (tharindu-bandara) [inline components/org.wso2.carbon.identity.context.rewrite.valve/src/main/java/org/wso2/carbon/identity/context/rewrite/valve/TenantContextRewriteValve.java:316]:
```suggestion
                }
                return errorPage;
            }
```
Need to break the flow here if the file is not available right?

Comment (GANGANI) [reply components/org.wso2.carbon.identity.context.rewrite.valve/src/main/java/org/wso2/carbon/identity/context/rewrite/valve/TenantContextRewriteValve.java:316]: If the file is not available, the default content will be used as the error page content without breaking the flow while logging the warning message.

File: components/org.wso2.carbon.identity.context.rewrite.valve/src/main/java/org/wso2/carbon/identity/context/rewrite/valve/TenantContextRewriteValve.java [hunk 2]
```diff
@@ -273,17 +280,47 @@ private void handleInvalidTenantDomainErrorResponse(Response response, int error
...
-        errorResponse.addProperty("code", HttpServletResponse.SC_FORBIDDEN);
-        errorResponse.addProperty("message", errorMsg);
-        errorResponse.addProperty("description", errorMsg);
-        response.getWriter().print(errorResponse.toString());
+    private String loadPageNotFoundErrorPage() {
+
+        String errorPage = "Page Not Found";
+        try {
+            Path pageNotFoundHtmlResponse =
+                    Paths.get(CarbonUtils.getCarbonHome(), "repository", "resources", "identity", "pages",
```
Comment (senthalan) [inline components/org.wso2.carbon.identity.context.rewrite.valve/src/main/java/org/wso2/carbon/identity/context/rewrite/valve/TenantContextRewriteValve.java:309]: It is better if we can load this in the component activate method and resolve the error page. Then reuse other than checking or each API calls.

Comment (GANGANI) [reply components/org.wso2.carbon.identity.context.rewrite.valve/src/main/java/org/wso2/carbon/identity/context/rewrite/valve/TenantContextRewriteValve.java]: Fixed with https://github.com/wso2-extensions/identity-carbon-auth-rest/pull/145/commits/a6ec2c2a744b12df63be45d834e8b7ec0cdafb8f

**Review by IsuraD — APPROVED** [review]

**Pull request merged**